WordPress/gutenberg
# Show all blocks all the times to avoid confusing UX

## What?


Explores #60991 

## Why?


Too often the inserter is empty, for instance when opening the inserter while a list block list item is selected or a button in a buttons block, the only available block is the list item , or the button respectively.

This is confusing because it's unclear why the other blocks are not available.

## How?


Try to show all the blocks all the time and insert blocks in the nearest possible location.


## Testing Instructions





1. In the site editor or the post editor add a list block
2. Add a list item
3. Open the inserter
4. **Notice all blocks are listed**
5. Insert a block that is _not_ the list item block, e.g. an image block
6. **Notice the image block is inserted right after the parent list block of the currently selected list item block**


### Testing Instructions for Keyboard


N/A

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/107534/ffd8aa77-f1fe-4be2-a3f3-77134bc5904a

# Known limitations

- the inserter is still empty when a block in a pattern with overrides is selected and no insertion is possible
- https://github.com/WordPress/gutenberg/pull/61873/files#r1611877561
